hathor/p2p/sync_v2/streamers.py

```diff
@@ -180,44 +184,67 @@ def send_next(self) -> None:
         # XXX: don't send the genesis or the current block
         if self.current_block is None or self.current_block.is_genesis:
             self.stop()
-            self.node_sync.send_blocks_end(StreamEnd.NO_MORE_BLOCKS)
+            self.sync_agent.send_blocks_end(StreamEnd.NO_MORE_BLOCKS)
             return
 
         self.schedule_if_needed()
 
 
 class TransactionsStreamingServer(_StreamingServerBase):
     """Streams all transactions confirmed by the given block, from right to left (decreasing timestamp).
+
+    If the start_from parameter is not empty, the BFS (Breadth-First Search) for the first block will commence
+    using the provided hashes. This mechanism enables streaming requests to continue from a specific point
+    should there be interruptions or issues.
     """
 
     def __init__(self,
-                 node_sync: 'NodeBlockSync',
+                 sync_agent: 'NodeBlockSync',
                  start_from: list[BaseTransaction],
                  first_block_hash: bytes,
                  last_block_hash: bytes,
                  *,
                  limit: int = DEFAULT_STREAMING_LIMIT) -> None:
         # XXX: is limit needed for tx streaming? Or let's always send all txs for
         # a block? Very unlikely we'll reach this limit
-        super().__init__(node_sync, limit=limit)
-
-        assert len(start_from) > 0
-        assert start_from[0].storage is not None
-        self.storage = start_from[0].storage
-        self.first_block_hash = first_block_hash
-        self.last_block_hash = last_block_hash
-        self.last_block_height = 0
-
-        self.bfs = BFSOrderWalk(self.storage, is_dag_verifications=True, is_dag_funds=True, is_left_to_right=False)
-        self.iter = self.bfs.run(start_from, skip_root=False)
+        super().__init__(sync_agent, limit=limit)
+
+        self.first_block: Block = cast(Block, self.tx_storage.get_transaction(first_block_hash))
+        self.last_block: Block = cast(Block, self.tx_storage.get_transaction(last_block_hash))
+        self.start_from = start_from
+
+        assert isinstance(self.first_block, Block)
+        assert isinstance(self.last_block, Block)
+
+        # TODO Validated start_from and use it when it's not empty.
```

Comment:
Just marking this TODO for bookkeeping

Reply:
The "use it" is done already. I just need to validate it properly.

Reply:
Done! https://github.com/HathorNetwork/hathor-core/pull/850/commits/93c2fb9b04d747d00a1524e8a2c59016c6a728b3